builder/vsphere/common/step_export.go

```diff
@@ -0,0 +1,225 @@
+//go:generate struct-markdown
+//go:generate mapstructure-to-hcl2 -type ExportConfig
+
+package common
+
+import (
+	"bytes"
+	"context"
+	"crypto/sha1"
+	"crypto/sha256"
+	"crypto/sha512"
+	"fmt"
+	"hash"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/hashicorp/packer/builder/vsphere/driver"
+	"github.com/hashicorp/packer/helper/multistep"
+	"github.com/hashicorp/packer/packer"
+	"github.com/pkg/errors"
+	"github.com/vmware/govmomi/nfc"
+	"github.com/vmware/govmomi/vim25/soap"
+	"github.com/vmware/govmomi/vim25/types"
+)
+
+var sha = map[int]func() hash.Hash{
+	1:   sha1.New,
+	256: sha256.New,
+	512: sha512.New,
+}
+
+type ExportConfig struct {
+	// name of the ovf. defaults to the name of the VM
+	Name string `mapstructure:"name"`
+	// overwrites existing ovf
+	Force bool `mapstructure:"force"`
+	// Include image files (*.{iso,img})
+	Images bool `mapstructure:"images"`
+	// Generate manifest using SHA 1, 256, 512 or 0 to skip
+	Sha       int          `mapstructure:"sha"`
+	OutputDir OutputConfig `mapstructure:",squash"`
+}
+
+type StepExport struct {
+	Name      string
+	Force     bool
+	Images    bool
+	Sha       int
+	OutputDir string
+	dest      string
+	mf        bytes.Buffer
+}
+
+func (s *StepExport) Cleanup(multistep.StateBag) {
+}
+
+func (s *StepExport) Run(ctx context.Context, state multistep.StateBag) multistep.StepAction {
+	ui := state.Get("ui").(packer.Ui)
+	vm := state.Get("vm").(*driver.VirtualMachine)
+
+	if s.Sha != 0 {
+		if _, ok := sha[s.Sha]; !ok {
+			state.Put("error", fmt.Errorf("unknown hash: sha%d", s.Sha))
+			return multistep.ActionHalt
+		}
+	}
+
+	if s.Name == "" {
+		info, err := vm.Info("name")
+		if err != nil {
+			state.Put("error", fmt.Errorf("unable to look up vm name"))
+			return multistep.ActionHalt
+		}
+		s.Name = info.Name
+	}
+	s.dest = filepath.Join(s.OutputDir, s.Name)
+
+	target := filepath.Join(s.dest, s.Name+".ovf")
+
+	if !s.Force {
```

Comment:
Same here -- this would definitely be better in a Prepare instead of failing at the end of a long run.

Reply:
moved this to the prepare